pkg/cmd/experimental/login/login.go

```diff
@@ -0,0 +1,62 @@
+package login
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd"
+	kubecmd "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd"
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/cmd/cli/cmd"
+	"github.com/openshift/origin/pkg/cmd/util/tokencmd"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func NewCmdLogin(name string, parent *cobra.Command) *cobra.Command {
+	clientConfig := defaultClientConfig(parent.PersistentFlags())
+	f := cmd.NewFactory(clientConfig)
+	f.BindFlags(parent.PersistentFlags())
+
+	cmds := &cobra.Command{
+		Use:   name,
+		Short: "Logs in to the OpenShift server and returns a session token",
+		Long: `Logs in to the OpenShift server and returns a session token
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			clientCfg, err := f.OpenShiftClientConfig.ClientConfig()
+			if err != nil {
+				glog.Fatal("%v\n", err)
+			}
+
+			usernameFlag := kubecmd.GetFlagString(cmd, "username")
+			passwordFlag := kubecmd.GetFlagString(cmd, "password")
+
+			accessToken, err := tokencmd.RequestToken(clientCfg, os.Stdin, usernameFlag, passwordFlag)
+			if err != nil {
+				glog.Fatal("%v\n", err)
+			}
+
+			fmt.Printf("Auth token: %v\n", string(accessToken))
+		},
+	}
+
+	cmds.Flags().StringP("username", "u", "", "Username, will prompt if not provided")
+	cmds.Flags().StringP("password", "p", "", "Password, will prompt if not provided")
+	return cmds
+}
+
+// Copy of kubectl/cmd/DefaultClientConfig, using NewNonInteractiveDeferredLoadingClientConfig
```

Comment:
It feels like we have this somewhere else in openshift as well.  Can you make a TODO to find and merge the duplicates?